components/nimbus/src/tests/helpers.rs

```diff
@@ -49,6 +48,50 @@ impl Default for NimbusTargetingHelper {
     }
 }
 
+/// A Rust implementation of the MetricsHandler trait
+/// Used to test recording of Glean metrics across the FFI within Rust
+///
+/// *NOTE: Use this struct's `new` method when instantiating it to lock the Glean store*
+#[derive(Clone)]
+pub struct TestMetrics {
+    state: Arc<Mutex<HashMap<String, serde_json::Value>>>,
+}
+
+impl TestMetrics {
+    pub fn new() -> Self {
+        TestMetrics {
+            state: Default::default(),
+        }
+    }
+
+    pub fn assert_get_vec_value(&self, key: &str) -> serde_json::Value {
+        self.state.lock().unwrap().get(key).unwrap().clone()
+    }
+}
+
+impl MetricsHandler for TestMetrics {
+    fn record_enrollment_statuses(&self, enrollment_status_extras: Vec<EnrollmentStatusExtraDef>) {
+        let key = "enrollment_status";
+        let mut state = self.state.lock().unwrap();
+        let new = serde_json::to_value(enrollment_status_extras).unwrap();
+        match state.get(key) {
+            Some(v) => {
+                let new_value = v
+                    .as_array()
+                    .unwrap()
+                    .iter()
+                    .chain(new.as_array().unwrap())
+                    .cloned()
+                    .collect::<Vec<serde_json::Value>>();
+                state.insert(key.into(), new_value.into());
+            }
+            None => {
+                state.insert(key.into(), new);
+            }
+        };
```

Comment:
Prefer the entry API to do only a single lookup.

```rust
map.entry(key)
   .and_modify(|v| /* ... */)
   .or_insert_with(|| /* ... */)
```

Reply:
Oooooh this API is news to me — this is super cool!